content/blog/2021/08/2021-02-08.adoc

```diff
@@ -0,0 +1,167 @@
+
+With this plugin, Jenkins can send and receive CloudEvents-compliant events to and from a wide variety of CI/CD tools using CloudEvents as their event format. This plugin makes chaining Jenkins with multiple tools like Tekton, Keptn, Knative and more, very easy. 
+
+
+== GSoC Phase 1 - CloudEvents Plugin
+
+=== Using CloudEvents plugin for Jenkins
+
+This plugin allows Jenkins to be configured as a source and sink, which can emit and consume CloudEvents from a range of tools simultaneously. 
+
+==== Jenkins as a Source
+Configuring Jenkins as a Source enables Jenkins to send CloudEvents to a CloudEvents sink. For Phase-I of this project, there is support for HTTP Sinks, however CloudEvents supports various link:https://github.com/cloudevents/spec#cloudevents-documents[protocol bindings]. Moving forward, there will also be support for other protocol bindings supported by CloudEvents. 
+
+
+To use Jenkins as a Source, the following configuration is needed:
+
+1. Click on *Manage Jenkins* in the Root-Actions menu on the left.
+image:https://github.com/ShrutiC-git/cloudevents-plugin/blob/main/public/manage_jenkins.png[Manage Jenkins]
+
+2. Inside the Manage Jenkins UI, search for *Configure System* under *System Configuration*.
+image:https://github.com/ShrutiC-git/cloudevents-plugin/blob/main/public/configure_system.png[Configure System]
+
+3. In the *Configure System* UI, sroll down to the CloudEvents plugin section, and this is where all the plugin configuration will be present. Here, you will have to enter the following information:
+
+* Sink Type (For now, HTTP Protocol Binding for CloudEvent and HTTP Sink is supported.)
+* Sink URL (URL of the Sink where you want the cloudevents sent.)
+* Events you want sent to the CloudEvents sink URL. 
+image:https://github.com/ShrutiC-git/cloudevents-plugin/blob/main/public/sinkType.png[Configure Sink to Receive Events]
+
+With Jenkins as a Source configured, Jenkins will send a POST request to the configured sink right as the selected event happens inside Jenkins. Each event has different data specific to the type of the event that is emitted. 
+
+
+==== Event Types, Payload and Metadata
+
+CloudEvents emitted by Jenkins follow the Binary-structure supported by CloudEvents, where the CloudEvents metadata is present inside the header, and the event-data is serialized as JSON, and present under request-body. This is the HTTP Protocol Binding for CloudEvents. Each protocl binding for CloudEvents follows definition specific to the binding protocol. 
+
+For now, the following Jenkins events are supported in the CloudEvents Plugin-Jenkins as a Source: 
+
+* link:https://github.com/ShrutiC-git/cloudevents-plugin#queue-events[**Queue Events**]
```

Comment:
Change the links to point to https://github.com/jenkinsci/cloudevents-plugin rather than your own personal fork. This should ensure a longer life to the blog post.